tests/codeql/build.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Install dependencies
+sudo apt install -y ninja-build libunwind-dev libboost-fiber-dev libssl-dev \
+    autoconf-archive libtool cmake g++ libzstd-dev bison libxml2-dev
+
+# Clone the Dragonfly repository
+git clone --recursive https://github.com/dragonflydb/dragonfly && cd dragonfly
```

Comment:
why do you need to clone the directory? does not it defeat the purpose of checking a PR?

seems that steps 
https://github.com/dragonflydb/dragonfly/pull/4034/files#diff-12783128521e452af0cfac94b99b8d250413c516ec71fe6d97dbea666ff7ba27R32-R33

should suffice